WrenchFu/gui.lua

```diff
@@ -11,52 +11,65 @@ Tweakable:
 ]]
 
 if not WrenchFu then WrenchFu = {} end
-WrenchFu.max_distance = 6
 
 
-function WrenchFu.open_gui_for(player_index, entity_name, position)
-    local reg = WrenchFu.handler_for(entity_name)
-
-    global.open_guis[player_index] = {
-        ["entity_name"] = entity_name,
-        ["position"] = position,
+function WrenchFu.open_gui_for(player_index, entity)
+    local reg = WrenchFu.handler_for(entity.name)
+    global.open_guis[player_index] = global.open_guis[player_index] or {}
+    
+    global.open_guis[player_index][entity.name] = {
+        ["position"] = entity.position,
         ["handler"] = reg,
     }
-
-    remote.call(reg.mod_name, reg.show_method, player_index, entity_name, position)
+    if not reg.legacy then
+        remote.call(reg.mod_name, reg.show_method, player_index, entity)
+    elseif reg.legacy=="0.1.2" then
+        remote.call(reg.mod_name, reg.show_method, player_index, entity.name, entity.position)
+    end
 end
 
 
-function WrenchFu.close_gui_for(player_index)
-    if global.open_guis[player_index] == nil then return end
-    local gui_data = global.open_guis[player_index]
+function WrenchFu.close_gui_for(player_index,entity_name)
+    if global.open_guis[player_index] == nil then return true end
+    if global.open_guis[player_index][entity_name] == nil then return true end
+    local gui_data = global.open_guis[player_index][entity_name]
 
     local reg = gui_data.handler
-    remote.call(reg.mod_name, reg.hide_method, player_index, gui_data.entity_name, gui_data.position)
-
-    global.open_guis[player_index] = nil
+    local permit= remote.call(reg.mod_name, reg.hide_method, player_index, entity_name, gui_data.position)
```

Comment:
Some guis  might have complex info or state stored in them, in such cases it's better to give client mod ability to raise confirmation dialogue rather than simply killing it. (I've been doing one such important gui recently.)
That's also the reason, the max_range is per entity type here.